Endpoint is not available on object store status causing a crash on the operator

<!-- **Are you in the right place?**
1. For issues or feature requests, please create an issue in this repository.
2. For general technical and non-technical questions, we are happy to help you on our [Rook.io Slack](https://slack.rook.io/).
3. Did you already search the existing open issues for anything similar? -->

**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
I was digging through an operator log in the CI and stumbled upon this crash:

```
E0106 21:14:39.858971       6 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 639 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x2204ba0, 0x3e91d50)
	/home/rook/go/pkg/mod/k8s.io/apimachinery@v0.20.0/pkg/util/runtime/runtime.go:74 +0x92
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/home/rook/go/pkg/mod/k8s.io/apimachinery@v0.20.0/pkg/util/runtime/runtime.go:48 +0x82
panic(0x2204ba0, 0x3e91d50)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/rook/rook/pkg/operator/ceph/object/user.(*ReconcileObjectStoreUser).initializeObjectStoreContext(0xc0001002a0, 0xc00161d040, 0x8, 0x0)
	/home/rook/go/src/github.com/rook/rook/pkg/operator/ceph/object/user/controller.go:313 +0x100
github.com/rook/rook/pkg/operator/ceph/object/user.(*ReconcileObjectStoreUser).reconcile(0xc0001002a0, 0xc0017030f0, 0x8, 0xc0017030d4, 0x9, 0xc000c08380, 0xc000e07c40, 0xc37be7, 0xc00026e550)
	/home/rook/go/src/github.com/rook/rook/pkg/operator/ceph/object/user/controller.go:191 +0x445
github.com/rook/rook/pkg/operator/ceph/object/user.(*ReconcileObjectStoreUser).Reconcile(0xc0001002a0, 0xc0017030f0, 0x8, 0xc0017030d4, 0x9, 0xc000c08380, 0xc0006657a8, 0x2b08420, 0xc0006657a0)
	/home/rook/go/src/github.com/rook/rook/pkg/operator/ceph/object/user/controller.go:129 +0x66
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00070b9e0, 0x22c4580, 0xc001419a40, 0x4f7319ac382c4700)
	/home/rook/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.3/pkg/internal/controller/controller.go:244 +0x27d
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00070b9e0, 0x0)
	/home/rook/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.3/pkg/internal/controller/controller.go:218 +0xcb
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc00070b9e0)
	/home/rook/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.3/pkg/internal/controller/controller.go:197 +0x2b
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000cd3970)
	/home/rook/go/pkg/mod/k8s.io/apimachinery@v0.20.0/pkg/util/wait/wait.go:155 +0x5e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000cd3970, 0x2ae5480, 0xc000c06360, 0x1, 0xc000165c80)
	/home/rook/go/pkg/mod/k8s.io/apimachinery@v0.20.0/pkg/util/wait/wait.go:156 +0xa3
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000cd3970, 0x3b9aca00, 0x0, 0x1, 0xc000165c80)
	/home/rook/go/pkg/mod/k8s.io/apimachinery@v0.20.0/pkg/util/wait/wait.go:133 +0xaa
k8s.io/apimachinery/pkg/util/wait.Until(0xc000cd3970, 0x3b9aca00, 0xc000165c80)
	/home/rook/go/pkg/mod/k8s.io/apimachinery@v0.20.0/pkg/util/wait/wait.go:90 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/home/rook/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.3/pkg/internal/controller/controller.go:179 +0x44d
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x1e0a0e0]
```

It looks like the status is not yet initialized when this line is called:
```golang
r.objContext.Endpoint = store.Status.Info["endpoint"]
```

**Expected behavior:**
The endpoint needs to be initialized reliably.

**How to reproduce it (minimal and precise):**
<!-- Please let us know any circumstances for reproduction of your bug. -->
Create an object store and observe if the operator crashes and restarts.

--- a/pkg/operator/ceph/object/user/controller.go
+++ b/pkg/operator/ceph/object/user/controller.go
@@ -310,12 +310,18 @@ func (r *ReconcileObjectStoreUser) initializeObjectStoreContext(u *cephv1.CephOb
 	}
 
 	r.objContext = objContext
-	r.objContext.Endpoint = store.Status.Info["endpoint"]
-	if store.Status != nil && store.Status.Info["secureEndpoint"] != "" {
-		r.objContext.Endpoint = store.Status.Info["secureEndpoint"]
+	if store.Status == nil {
+		return errors.New("failed to initialize ceph object user because unknown object store status")
 	}
-	if r.objContext.Endpoint == "" {
+
+	if _, ok := store.Status.Info["endpoint"]; ok {
+		r.objContext.Endpoint = store.Status.Info["endpoint"]
+	} else {
 		return errors.Errorf("endpoint is missing in the status Info")
+	}
+
+	if _, ok := store.Status.Info["secureEndpoint"]; ok {
+		r.objContext.Endpoint = store.Status.Info["secureEndpoint"]
 	}
 
 	return nil

